Makefile

```diff
@@ -90,29 +92,37 @@ install-mockgen:
 	go install go.uber.org/mock/mockgen@latest
 
 install-golangci-lint:
-	@which golangci-lint || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.62.0
+	@command -v golangci-lint >/dev/null 2>&1 || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.62.0
```

Comment:
what is difference ?

Reply:
<h3>Key Differences</h3><ol><li><p><strong><code>which</code> Command</strong>:</p><ul><li><strong>Code</strong>:<pre class="!overflow-visible"><div class="contain-inline-size rounded-md border-[0.5px] border-token-border-medium relative bg-token-sidebar-surface-primary dark:bg-gray-950"><div class="flex items-center text-token-text-secondary px-4 py-2 text-xs font-sans justify-between rounded-t-md h-9 bg-token-sidebar-surface-primary dark:bg-token-main-surface-secondary select-none">makefile</div><div class="sticky top-9 md:top-[5.75rem]"><div class="absolute bottom-0 right-2 flex h-9 items-center"><div class="flex items-center rounded bg-token-sidebar-surface-primary px-2 font-sans text-xs text-token-text-secondary dark:bg-token-main-surface-secondary"><span class="" data-state="closed"><button class="flex gap-1 items-center select-none py-1"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" class="icon-sm"><path fill-rule="evenodd" clip-rule="evenodd" d="M7 5C7 3.34315 8.34315 2 10 2H19C20.6569 2 22 3.34315 22 5V14C22 15.6569 20.6569 17 19 17H17V19C17 20.6569 15.6569 22 14 22H5C3.34315 22 2 20.6569 2 19V10C2 8.34315 3.34315 7 5 7H7V5ZM9 7H14C15.6569 7 17 8.34315 17 10V15H19C19.5523 15 20 14.5523 20 14V5C20 4.44772 19.5523 4 19 4H10C9.44772 4 9 4.44772 9 5V7ZM5 9C4.44772 9 4 9.44772 4 10V19C4 19.5523 4.44772 20 5 20H14C14.5523 20 15 19.5523 15 19V10C15 9.44772 14.5523 9 14 9H5Z" fill="currentColor"></path></svg>Copy code</button></span></div></div></div><div class="overflow-y-auto p-4" dir="ltr"><code class="!whitespace-pre hljs language-makefile">@which golangci-lint || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.62.0
</code></div></div></pre></li><li><strong>Behavior</strong>:<ul><li><code>which golangci-lint</code> checks if <code>golangci-lint</code> is in the system's PATH.</li><li>If not found, <code>which</code> exits with a non-zero exit status, triggering the <code>go install</code> command.</li></ul></li><li><strong>Portability</strong>:<ul><li>The <code>which</code> command is not built into all shells, so it might not work in environments where <code>which</code> is unavailable.</li></ul></li></ul></li><li><p><strong><code>command -v</code> Built-in</strong>:</p><ul><li><strong>Code</strong>:<pre class="!overflow-visible"><div class="contain-inline-size rounded-md border-[0.5px] border-token-border-medium relative bg-token-sidebar-surface-primary dark:bg-gray-950"><div class="flex items-center text-token-text-secondary px-4 py-2 text-xs font-sans justify-between rounded-t-md h-9 bg-token-sidebar-surface-primary dark:bg-token-main-surface-secondary select-none">makefile</div><div class="sticky top-9 md:top-[5.75rem]"><div class="absolute bottom-0 right-2 flex h-9 items-center"><div class="flex items-center rounded bg-token-sidebar-surface-primary px-2 font-sans text-xs text-token-text-secondary dark:bg-token-main-surface-secondary"><span class="" data-state="closed"><button class="flex gap-1 items-center select-none py-1"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" class="icon-sm"><path fill-rule="evenodd" clip-rule="evenodd" d="M7 5C7 3.34315 8.34315 2 10 2H19C20.6569 2 22 3.34315 22 5V14C22 15.6569 20.6569 17 19 17H17V19C17 20.6569 15.6569 22 14 22H5C3.34315 22 2 20.6569 2 19V10C2 8.34315 3.34315 7 5 7H7V5ZM9 7H14C15.6569 7 17 8.34315 17 10V15H19C19.5523 15 20 14.5523 20 14V5C20 4.44772 19.5523 4 19 4H10C9.44772 4 9 4.44772 9 5V7ZM5 9C4.44772 9 4 9.44772 4 10V19C4 19.5523 4.44772 20 5 20H14C14.5523 20 15 19.5523 15 19V10C15 9.44772 14.5523 9 14 9H5Z" fill="currentColor"></path></svg>Copy code</button></span></div></div></div><div class="overflow-y-auto p-4" dir="ltr"><code class="!whitespace-pre hljs language-makefile">@command -v golangci-lint &gt;/dev/null 2&gt;&amp;1 || go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.62.0
</code></div></div></pre></li><li><strong>Behavior</strong>:<ul><li><code>command -v golangci-lint</code> checks if <code>golangci-lint</code> is in the system's PATH using a shell built-in.</li><li>It outputs nothing to <code>stdout</code> (because of <code>&gt;/dev/null</code>) and suppresses error messages (via <code>2&gt;&amp;1</code>).</li><li>If the command does not exist, it exits with a non-zero status, triggering the <code>go install</code> command.</li></ul></li><li><strong>Portability</strong>:<ul><li><code>command -v</code> is POSIX-compliant and more portable across different systems compared to <code>which</code>.</li></ul></li></ul></li></ol><h3>Summary of Pros and Cons</h3>

Feature | which | command -v
-- | -- | --
Portability | Less portable (not always available) | More portable (POSIX-compliant)
Shell Dependency | External command | Built into most shells
Error Suppression | No built-in suppression mechanism | Can suppress both stdout and stderr easily
Preferred Use Case | For environments where which exists | General-purpose across platforms



<h3></h3>